### DISCUSSION TITLE
Write replaygain_album_gain instead of REPLAYGAIN_ALBUM_GAIN

### DISCUSSION THREAD

**Discussion by tomorbit** [beetbox/beets#4500] [Q&A]
Beets seems to write replaygain tags in upper-case.
There are music player in the wild which only understand lower-case replaygain tags (Poweramp, KODI).

Is it possible to write lower-case replaygain tags instead of upper-case tags with beets?
For example replaygain_album_gain instead of REPLAYGAIN_ALBUM_GAIN ?

**Comment by sampsyo** [comment]
This would be an issue for [MediaFile](https://github.com/beetbox/mediafile), which is our tag-name abstraction layer. However, it actually does already store *both* tag names, as you can see here:
https://github.com/beetbox/mediafile/blob/edc64145e9e96624f8a77fd3cb929891c946d4dc/mediafile.py#L2202-L2209

Perhaps the problem with those two tools is elsewhere?

**Reply by tomorbit** [reply]
I can only find upper-case replay gain tags in my beets tagged files with a hex editor.

Maybe it may be useful to go through my whole conversion workflow:

First of all I download a clean WAV file from bandcamp.com for example. I convert the WAV file to a mp3 via Foobar using the Lame mp3 encoder. After this I get a mp3 file: 01 - New Light.mp3. This is completely without any tags.

After that this mp3 goes into Picard. After that I get a tagged mp3 file: 02 - New Light.mp3. This is without replay gain tags.

This goes through processing by Beets. For this I use a Docker Beets version (https://hub.docker.com/r/linuxserver/beets). After that I get a mp3 file including replay gain tags: 03 - New Light.mp3

Using a hex editor I only find upper-case Replay Gain tags in the header of the file.

Did I do anything wrong in the process? Is the Docker version of Beets the problem?

[01 - New Light.zip](https://github.com/beetbox/beets/files/9668804/01.-.New.Light.zip)
[02 - New Light.zip](https://github.com/beetbox/beets/files/9668806/02.-.New.Light.zip)
[03 - New Light.zip](https://github.com/beetbox/beets/files/9668812/03.-.New.Light.zip)
![03 - New Light - Header](https://user-images.githubusercontent.com/5084469/192885491-bf485e2e-22b7-48fb-8386-0ed8d8900031.png)

**Reply by sampsyo** [reply]
I don't see anything obviously wrong here! I might try `beet write` to make sure the tags are rewritten, and also making sure that your system has a version of MediaFile with the lines of code I linked to above.

**Reply by tomorbit** [reply]
I tried `beet write` (files have not changed) and `beet write -f` (files have changed) but the mp3 files still only have upper-case Replay Gain tags. I compared the file mediafile.py in Docker with the provided version from above. They seem to be the same in the specified part.

![mediafile-col2171](https://user-images.githubusercontent.com/5084469/193118285-42dea43c-a823-45fa-9c28-ab68046495c1.png)

**Reply by sampsyo** [reply]
That's odd! I don't have a good explanation for why this could be happening. Maybe it would be helpful to try removing the metadata and re-adding it—perhaps with the `scrub` plugin?

**Reply by tomorbit** [reply]
What I tried: I have completely deleted the ReplayGain tags from a series of files with `zero `and checked that they are really gone in the library (`info`) and the mp3 files. After that I re-tagged them with replaygain (`replaygain`). After that I found the ReplayGain tags again, only in upper-case.

**Reply by sampsyo** [reply]
Terribly weird! Thanks for trying all those things… I have to admit I'm stumped at this point. I don't see why this would be misbehaving in this way. I'm honestly not sure what to try next!

**Reply by tomorbit** [reply]
I tried something else.
I changed the lines of code in mediafile.py to see what tag is written. After applying `scrub` and `relaygain` again, I suddenly found two tags in my file, although with different names.
So when both are named differently, all of a sudden both show up. If both have the same name but are capitalized differently, they don't show up. Is this a bug?

![Clipboard 1](https://user-images.githubusercontent.com/5084469/194117912-0c47220f-0332-4673-b2b3-2ef49e6a7eb8.png)
![Clipboard 2](https://user-images.githubusercontent.com/5084469/194117932-44bebc12-37e3-46a3-ac4a-eabd21f8424a.png)

**Reply by sampsyo** [reply]
Wow; that is very interesting! It may indeed. If it is, it would be a bug in [MediaFile](https://github.com/beetbox/mediafile) or the underlying Mutagen library. Any chance you could file a bug there with instructions about how to reproduce the problem?

**Reply by tomorbit** [reply]
sure! -> done!
https://github.com/beetbox/mediafile/issues/62
